### ISSUE TITLE
Create 'term' filter for Admin Panel

### ISSUE THREAD

**Issue by bryanc208** [uwblueprint/website#84]
Currently, the admin panel collects applications for both students with no distinction for the term that we accepted applications in. This causes a problem for us when we want to reuse the admin panel for future terms as it becomes very difficult to differentiate between the applications. It also causes a lot of clutter as we ideally do not ever want to delete any applications in case we want them for future reference.

The fix is to create a `term` filter for the admin panel. The idea is to fix a 'term' field on all student applications as a hidden form element. This `term` field would have the current school term (according to UWaterloo nomenclature i.e.) W17, S18, F19, etc.) and it would allow us to group the applications by term. 

We should also consider ways we can utilize external environment variables so that we do not need to push out a new update to the website each time we want to start off a new recruiting term. This is lower priority.

Subtasks under this issue list all of the steps to getting this completed.

The deadline for this task is roughly July 1, 2017 so that we can prepare for F17 applications.

- [ ] Create a hidden form element that fixes the current term for which applications are being accepted 
- [ ] Create a filter by `term` on the admin panel as a drop down selection. The default should always select the newest term. (This can be a little tricky to do with strings representing the terms. Perhaps consider using enums for this)
- [ ] Migrate all of the existing applications to have the `term=W17`

**Comment by bryanc208** [comment]
This depends on Issue #83
